packages/xrpl/src/models/transactions/common.ts

```diff
@@ -455,13 +457,16 @@ export function validateCredentialType(tx: Record<string, unknown>): void {
  * @param credentials An array of credential IDs to check for errors
  * @param transactionType The transaction type to include in error messages
  * @param isStringID Toggle for if array contains IDs instead of AuthorizeCredential objects
+ * @param maxLengthCredentialsArray The maximum length of the credentials array.
+ *        PermissionedDomainSet transaction uses 10, other transactions use 8.
  * @throws Validation Error if the formatting is incorrect
  */
 // eslint-disable-next-line max-lines-per-function -- separating logic further will add unnecessary complexity
 export function validateCredentialsList(
   credentials: unknown,
   transactionType: string,
   isStringID: boolean,
+  maxLengthCredentialsArray: number = MAX_CREDENTIALS_LIST_LENGTH,
```

Comment:
This shouldn't have a default value

Reply:
Do you expect this method to be used in other contexts? It has been predominantly used in the CredentialCreate validation

Reply:
Neither option really feels like a "default", and I wouldn't want a library developer to make a mistake in the future (and use the wrong value) because of that.

Reply:
ok, fixed in c619920a